src/libclipper/src/redis.cpp

```diff
@@ -313,41 +366,48 @@ std::unordered_map<std::string, std::string> get_application_by_key(
 }
 
 void subscribe_to_keyspace_changes(
-    int db, Subscriber& subscriber,
+    int db, std::string prefix, Subscriber& subscriber,
     std::function<void(const std::string&, const std::string&)> callback) {
   std::ostringstream subscription;
-  subscription << "__keyspace@" << std::to_string(db) << "__:*";
+  subscription << "__keyspace@" << std::to_string(db) << "__:" << prefix << "*";
   std::string sub_str = subscription.str();
   log_info_formatted(LOGGING_TAG_REDIS, "SUBSCRIPTION STRING: {}", sub_str);
-  subscriber.psubscribe(
-      sub_str, [callback](const std::string& topic, const std::string& msg) {
-        size_t split_idx = topic.find_first_of(":");
-        std::string key = topic.substr(split_idx + 1);
-        log_info_formatted(LOGGING_TAG_REDIS, "MESSAGE: {}", msg);
-        callback(key, msg);
-      });
+  subscriber.psubscribe(sub_str, [callback, prefix](const std::string& topic,
+                                                    const std::string& msg) {
+    size_t split_idx = topic.find_first_of(":");
+    std::string key = topic.substr(split_idx + 1 + prefix.size());
+    log_info_formatted(LOGGING_TAG_REDIS, "MESSAGE: {}", msg);
+    callback(key, msg);
+  });
 }
 
 void subscribe_to_model_changes(
     Subscriber& subscriber,
     std::function<void(const std::string&, const std::string&)> callback) {
-  subscribe_to_keyspace_changes(REDIS_MODEL_DB_NUM, subscriber,
+  subscribe_to_keyspace_changes(REDIS_MODEL_DB_NUM, "", subscriber,
                                 std::move(callback));
 }
 
 void subscribe_to_container_changes(
     Subscriber& subscriber,
     std::function<void(const std::string&, const std::string&)> callback) {
-  subscribe_to_keyspace_changes(REDIS_CONTAINER_DB_NUM, subscriber,
+  subscribe_to_keyspace_changes(REDIS_CONTAINER_DB_NUM, "", subscriber,
```

Comment:
Hmm I actually disagree here. Defining NO_PREFIX as a constant seems like treating this as a special case. I'll define a local variable `std::string prefix = ""` in each method instead.